source/agora/utils/HumanReadable.d

```diff
@@ -0,0 +1,284 @@
+/*******************************************************************************
+
+    Defines method to format types to their human readable representation
+
+    This approach allows to take the string formatting out of the type,
+    simplifying the implementation and reducing dependencies.
+    It is assumed that a human-readable representation will not rely on
+    non-visible data, or data that cannot be accessed without mutation.
```

Comment:
Could you elaborate what "non-visible data" means?

Reply:
`private` data. Because nowadays D is using "visibility" rather than "protection".

Reply:
Well I'll add it to the docs